### PR TITLE
Bugfix FXIOS-5728 [v113] Clear history doesn't clear recently visited

### PR DESCRIPTION
**Pull request by rilury**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5728)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13186)

### Description
This PR ensures Recently visited section gets refreshed after deleting history using delete history metadata method which is already implemented and sends a notification that allows the UI section for recently visited to update its UI and data.

### Pull requests checks where applicable
- [X] Fill in the three TODOs above (tickets number and description of your work)
- [X] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Utils/HistoryDeletionUtility.swift [hunk 1]
```diff
@@ -112,6 +112,12 @@ class HistoryDeletionUtility: HistoryDeletionProtocol {
         profile.places.deleteHistoryMetadata(since: dateInMilliseconds) { _ in }
     }
 
+    func deleteHistoryMetadataOlderThan(_ dateOption: HistoryDeletionUtilityDateOptions) {
```
Comment (yoanarios) [inline Client/Utils/HistoryDeletionUtility.swift:115]: Could be possible to add unit test for this new function in `HistoryDeletionUtilityTests`?

**Review by yoanarios — APPROVED** [review]

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 30.19%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 2 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 27.04
| File | Coverage ||
| --- | --- | --- |
ClearHistorySheetProvider.swift | 2.8% | ⚠️
HistoryDeletionUtility.swift | 95.42% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 58b6f01bf87b0a6e960e8cc3c02849a830efc72f
</p>

**Review by adudenamedruby — APPROVED** [review]

File: Client/Utils/HistoryDeletionUtility.swift (hunk 1 above)
Comment (rilury) [reply Client/Utils/HistoryDeletionUtility.swift:115]: Hey, @yoanarios, I updated the PR with unit test for this function applicable in all scenarios depending on timeframe. It should be good to go now. Thank you so much!

**Comment by yoanarios** [comment]
@rilury there is a swiftlint warning [here]( https://github.com/mozilla-mobile/firefox-ios/pull/13744/checks?check_run_id=12335279065). I will merge the PR once is fix 🙏

**Comment by rilury** [comment]
> @rilury there is a swiftlint warning [here](https://github.com/mozilla-mobile/firefox-ios/pull/13744/checks?check_run_id=12335279065). I will merge the PR once is fix 🙏

done! thank you 😊

**Comment by yoanarios** [comment]
Merging without running Bitrise, it was green before the Swiftlint commit

**Pull request merged**